### ISSUE TITLE
Loop argument on Video, Audio and OffthreadVideo

### ISSUE THREAD

**Issue by B3nnjoe** [remotion-dev/remotion#1300]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%24167%20Rewarded-%237E24E3.svg" />](https://issuehunt.io/r/remotion-dev/remotion/issues/1300)

**🎃 This issue is part of our Hacktoberfest campaign!**
📚 [Read more about Hacktoberfest here](https://remotion.dev/hacktoberfest)
🕛 This issue is currently assigned to @DerrykBoyd 

# Feature Request 🛍️
A loop argument on `<Video/>` and `<OffthreadVideo>`
Since now you need to use the `<Loop/>` and it needs to re-download the video for every iteration. (Correct me if im wrong.)

## Acceptance criteria

- Accept a prop `loop` on the `<Video>`, `<OffthreadVideo>` and `<Audio>` components.
- It should work for development and rendering versions of the component.
- Mount the video/audio tag as normal and once the video is ready, get it's duration using a ref.
- Once the duration is read, use recursion - if the loop property is there, it should render the same component again but without the loop property. For example: `<Video loop src="video.mp3" />` will become `<Loop><Video src="video.mp3" durationInFrames={xxx}/></Loop>`
- Don't use the APIs in `@remotion/media-utils` - the components in `remotion` should have no dependencies.
- It should also work if the `src` of a video/audio changes!
- It should use the `delayRender()` / `continueRender()` pattern to ensure that the duration is determined before the render starts.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $167.00)

- [<img src='https://avatars.githubusercontent.com/u/1629785?v=4' alt='jonnyburger' width=24 height=24> jonnyburger](https://issuehunt.io/u/jonnyburger) ($167.00)

### Submitted pull Requests
- [#1382 Add loop prop for Video and Audio components](https://issuehunt.io/r/remotion-dev/remotion/pull/1382)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/remotion-dev/remotion/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by JonnyBurger** [comment]
Thanks for filing, this is a valid issue!
It will not be slower if you use Loop though, Remotion will not download it multiple times. Still I think it would be a nicer API!

**Comment by DerrykBoyd** [comment]
Hi @JonnyBurger 

I'd like to take this issue on! Thanks

**Comment by JonnyBurger** [comment]
@DerrykBoyd Nice, good luck! Assigned you

**Comment by Pompette** [comment]
I'd love to work on this if @DerrykBoyd doesn't have the time to do it, let me know :) !

**Comment by DerrykBoyd** [comment]
Hey @JonnyBurger 

I tried this out tonight, but I ended up getting a little stuck. I may need some help.

I tried to follow the general steps in the description and I can get the video looping, but there seems to be two videos playing and one just overlays on top showing the last frame all the time. 

What's the best way for me to share my work and get feedback? Just open up a draft PR maybe? Even though I'm nowhere near completing this.
